File: api/user-service/src/main/java/com/carrental/userservice/model/dto/PasswordResetDTO.java
package com.carrental.userservice.model.dto;

import com.carrental.userservice.annotation.PasswordMatches;
import com.carrental.userservice.annotation.ValidPassword;
import lombok.Data;
import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.NotEmpty;

@Data
@PasswordMatches
public class PasswordResetDTO {

    @NotEmpty
    @ValidPassword
    @Length(max = 16)
    private String newPassword;

    @NotEmpty
    @Length(max = 16)
    private String confirmPassword;

    @NotEmpty
    private String token;
}

File: frontend-app/shared/model/PasswordResetDTO.ts
export class PasswordResetDTO {
    newPassword: string;
    confirmPassword: string;
    token: string;

    constructor(newPassword: string, confirmPassword: string, token: string) {
        this.newPassword = newPassword;
        this.confirmPassword = confirmPassword;
        this.token = token;
    }
}
